### PR TITLE
Add support for `ListBlock`

### LINKED ISSUE #22: Translatable ListBlock
We can't make `ListBlock` translatable because we need a way to uniquely identify blocks on different translations of a page while not being sensitive to ordering changes, deleting and adding new items.

The best alternative at the moment is to use a `StreamBlock` with a single block type, but the UI for this isn't very nice.

We can probably come up with a replacement for `ListBlock` that has the exact same UI and API, but the database representation is a list of objects with ``uuid`` and ``value`` keys.

### PR DESCRIPTION
**Pull request by zerolab**
Builds on #509 (as it needs Wagtail 2.16)

Fixes #22

### CODE REVIEW

**Review by kaedroho — CHANGES_REQUESTED** [review]
This looks good to me so far. I think we should add a check in to make sure that the blocks have IDs and not extract these

File: wagtail_localize/segments/extract.py [hunk 1]
```diff
@@ -133,8 +134,14 @@ def handle_struct_block(self, struct_block):
         return segments
 
     def handle_list_block(self, list_block):
-        # TODO
-        return []
+        segments = []
+        if WAGTAIL_VERSION >= (2, 16):
```
[the remark this answers is not in the document]
Comment (zerolab) [reply wagtail_localize/segments/extract.py:138]: not sure I understand your question.
the `ListBlock` UUID change in 2.16 should provide block IDs if there none, if I understood the PR and the changes correctly

[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail_localize/segments/extract.py:138]: > the ListBlock UUID change in 2.16 should provide block IDs if there none

I'm not sure if these would be persistent though, it looks like they would be made up on the fly. So the next time they extraction is done, we would be given a different set of IDs.

[the remark this answers is not in the document]
Comment (zerolab) [reply wagtail_localize/segments/extract.py:138]: ah, I see. will need to read into the core handling of it and see if there's a reliable way to check if they have or not IDs

[the remark this answers is not in the document]
Comment (zerolab) [reply wagtail_localize/segments/extract.py:138]: dev note: at the moment this is made tricky by https://github.com/wagtail/wagtail-localize/blob/main/wagtail_localize/segments/extract.py#L142-L146

[the remark this answers is not in the document]
Comment (zerolab) [reply wagtail_localize/segments/extract.py:138]: @kaedroho ok, found an OK-ish solution - 6d130e5
if you have any suggestions for improvement, would love to hear them

**Review by kaedroho — APPROVED** [review]
This looks good to me, thanks!

**Pull request merged**